File: java/src/main/java/gov/doc/ntia/sigmf/ext/annotation/emitter/EmitterAnnotation.java
package gov.doc.ntia.sigmf.ext.annotation.emitter;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import gov.doc.ntia.sigmf.Annotation;
import gov.doc.ntia.sigmf.ext.global.core.Antenna;
import gov.doc.ntia.sigmf.ext.global.waveform.Waveform;
import gov.doc.ntia.sigmf.serialization.DoubleSerializer;

import java.io.Serializable;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class EmitterAnnotation extends Annotation  {

    private static final long serialVersionUID = 1L;

    @JsonProperty(value="ntia-emitter:id", required = true)
    protected String id;


    @JsonProperty(value="ntia-emitter:waveform", required = false)
    protected Waveform waveform;


    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Waveform getWaveform() {
        return waveform;
    }

    public void setWaveform(Waveform waveform) {
        this.waveform = waveform;
    }

}

File: C#/sigmf/sigmf/ext/Annotationdir/emitter/EmitterAnnotation.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    [Serializable()]
    public class EmitterAnnotation : Annotation
    {
        private static readonly long serialVersionUID = 1L;

        [JsonProperty("ntia-emitter:id")]
        protected string id;

        [JsonProperty("ntia-emitter:waveform")]
        protected Waveform waveform;

        [JsonProperty("ntia-emitter:altitude")]
        protected double altitude;

        [JsonProperty("ntia-emitter:speed")]
        protected double speed;

        [JsonProperty("ntia-emitter:bearing")]
        protected double bearing;

        public string Id { get => id; set => id = value; }
        public Waveform Waveform { get => waveform; set => waveform = value; }
        public double Altitude { get => altitude; set => altitude = value; }
        public double Speed { get => speed; set => speed = value; }
        public double Bearing { get => bearing; set => bearing = value; }
    }
}
